### ISSUE TITLE
diamoand error

### ISSUE THREAD

**Issue by suresh2014** [moshi4/FastDTLmapper#67]
hi,
while using fastdtlmapper i got this error
then i installed diamond
thereafter also i am getting same error, kindly look into this error

thank you
-------------------------------

Checking required programs are installed
----------------------------------------

ERROR: Cannot run diamond
Format of make database command:
  diamond makedb --in INPUT -d OUTPUT
ERROR: Cannot run diamond
Format of search database command:
  diamond blastp -d DATABASE -q INPUT -o OUTPUT --more-sensitive -p 1 --quiet -e 0.001 --compress 1
Please check diamond is installed and that the executables are in the system path

ERROR: An error occurred, ***please review the error messages*** they may contain useful information about the problem.

# 02. Align each OG(Ortholog Group) sequences using mafft
0% 0:0=0s                                                                       
Finished: 2022/09/22 19:28:50 (Elapsed time = 0.000[h])

# 03. Trim each OG alignment using trimal
0% 0:0=0s                                                                       
Finished: 2022/09/22 19:28:50 (Elapsed time = 0.000[h])

# 04. Reconstruct each OG gene tree using iqtree
0% 0:0=0s                                                                       
Finished: 2022/09/22 19:28:51 (Elapsed time = 0.000[h])

# 05. Correct gene tree multifurcation using treerecs
0% 0:0=0s                                                                       
Finished: 2022/09/22 19:28:51 (Elapsed time = 0.000[h])

# 06. DTL reconciliation using AnGST
0% 0:0=0s                                                                       
Finished: 2022/09/22 19:28:51 (Elapsed time = 0.000[h])

# 07. Aggregate and map DTL reconciliation result
Traceback (most recent call last):
  File "/home/pbl2-ec/miniconda3/bin/FastDTLmapper", line 8, in <module>
    sys.exit(main())
  File "/home/pbl2-ec/miniconda3/lib/python3.9/site-packages/fastdtlmapper/scripts/FastDTLmapper.py", line 57, in main
    aggregate_and_map(args, outpath)
  File "/home/pbl2-ec/miniconda3/lib/python3.9/site-packages/fastdtlmapper/util/time.py", line 12, in wrap
    ret = func(*args, **kargs)
  File "/home/pbl2-ec/miniconda3/lib/python3.9/site-packages/fastdtlmapper/scripts/FastDTLmapper.py", line 245, in aggregate_and_map
    output_aggregate_map_results(outpath, group_id2all_node_event)
  File "/home/pbl2-ec/miniconda3/lib/python3.9/site-packages/fastdtlmapper/scripts/FastDTLmapper.py", line 332, in output_aggregate_map_results
    UtilTree.map_node_event(
  File "/home/pbl2-ec/miniconda3/lib/python3.9/site-packages/fastdtlmapper/util/tree.py", line 96, in map_node_event
    node_event = node_id2node_event[node.name]
KeyError: 'N001'

**Comment by moshi4** [comment]
It seems that the error is occurring at the first processing stage of FastDTLmapper, which estimates the ortholog group by OrthoFinder. The error is caused by the fact that diamond (v2.0.5), which is embedded in OrthoFinder, is not working properly.

I will give you v2.0.5 and the latest v2.0.15 of the diamond executable binary, so could you please check if the following command works in your environment?

```
./diamond_v2.0.5 makedb --in test.faa -d test_v2.0.5
./diamond_v2.0.15 makedb --in test.faa -d test_v2.0.15
```

If only diamond (v2.0.15) seems to work properly in your environment, I would like to replace v2.0.5 with v2.0.15 and release a new FastDTLmapper to resolve the cause of the error.

___

[diamond_run_test.zip](https://github.com/moshi4/FastDTLmapper/files/9528125/diamond_run_test.zip)
- diamond_v2.0.5
- diamond_v2.0.15
- test.faa

**Comment by suresh2014** [comment]
hi 
thanks for prompt  response
for me diamond_v2.0.15 working nicely

--------------------
## terminal output ##

(base) pbl2-ec@pbl2:~/diamond_run_test$ ./diamond_v2.0.5 makedb --in test.faa -d test_v2.0.5
terminate called after throwing an instance of 'std::runtime_error'
  what():  CPU does not support POPCNT. Please compile the software from source.
Aborted (core dumped)
(base) pbl2-ec@pbl2:~/diamond_run_test$ ./diamond_v2.0.15 makedb --in test.faa -d test_v2.0.15
diamond v2.0.15.153 (C) Max Planck Society for the Advancement of Science
Documentation, support and updates available at http://www.diamondsearch.org
Please cite: http://dx.doi.org/10.1038/s41592-021-01101-x Nature Methods (2021)

#CPU threads: 2
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Database input file: test.faa
Opening the database file...  [0s]
Loading sequences...  [0.002s]
Masking sequences...  [0.018s]
Writing sequences...  [0s]
Hashing sequences...  [0s]
Loading sequences...  [0s]
Writing trailer...  [0s]
Closing the input file...  [0s]
Closing the database file...  [0s]

Database sequences  686
  Database letters  229919
     Database hash  4f9ac375ba759867838d956b6530044c
        Total time  0.023000s
(base) pbl2-ec@pbl2:~/diamond_run_test$ 


---------------------

**Comment by moshi4** [comment]
I have released new FastDTLmapper v0.3.4 which replaces diamond with v2.0.15, please try it.

**Comment by suresh2014** [comment]
its working ..cool .. thanks
